presto-main/src/main/java/com/facebook/presto/execution/QueryManagerStats.java

```diff
@@ -145,6 +145,8 @@ private class StatisticsListener
         private boolean stopped;
         @GuardedBy("this")
         private boolean started;
+        @GuardedBy("this")
+        private boolean queued = true;
```

Comment:
The default value is set to true since this listener starts tracking state changes only after query is queued.